### ISSUE TITLE
False negative in 942230

### ISSUE THREAD

**Issue by NiceYouKnow** [coreruleset/coreruleset#2230]
### Describe the bug
This is the regularity of 942230:
```
(?i:[\s()]case\s+when.*?then|\)\s*?like\s*?\(|select.*?having\s*?[^\s]+\s*?[^\w\s]|if\s?\([\d\w]\s*?[=<>~])
```
It can be split into four independent rules:
```
1: (?i:[\s()]case\s+when.*?then)
2: (?i:\)\s*?like\s*?\()
3: (?i:select.*?having\s*?[^\s]+\s*?[^\w\s])
4: (?i:if\s?\([\d\w]\s*?[=<>~])
```
My question is that there may be a risk of missing the attack on the fourth rule.

Below are two payloads of the same type:
```
1: if(1=2, -1, 2) union select table_name from information_schema.tables limit 1
2: IF(7423=7424) SELECT 7423 ELSE DROP FUNCTION xcjl--
```
You can check the test in https://regex101.com/r/h9Vz4I/2, the regular rule only hits the first payload, but not the second payload.

So my suggestion for modification is：
```
(?i:if\s?\([\w]+\s*?[=<>~])
```
Remove `\d`, because `\w` already contains it.

The modification effect can be viewed https://regex101.com/r/k0G4Ux/1/

### Your Environment
CRS version v3.4/dev

* CRS version (e.g., v3.2.0):
* Paranoia level setting:
* ModSecurity version (e.g., 2.9.3):
* Web Server and version (e.g., apache 2.4.41):
* Operating System and version:

**Comment by lifeforms** [comment]
I am just peering through this issue, but in MySQL, the syntax for the CASE operator is different. The example would be correct as:

```
mysql> select case when 1=1 then 1 else 2 end;
+---------------------------------+
| case when 1=1 then 1 else 2 end |
+---------------------------------+
|                               1 |
+---------------------------------+
```

However, I don't know how other SQL dialects allow CASE.

**Comment by dune73** [comment]
@NiceYouKnow are you still interested in this? Could you respond to @lifeforms' question?

**Comment by NiceYouKnow** [comment]
> I am just peering through this issue, but in MySQL, the syntax for the CASE operator is different. The example would be correct as:
> 
> ```
> mysql> select case when 1=1 then 1 else 2 end;
> +---------------------------------+
> | case when 1=1 then 1 else 2 end |
> +---------------------------------+
> |                               1 |
> +---------------------------------+
> ```
> 
> However, I don't know how other SQL dialects allow this. What servers accept your notation?

Your example is correct, no problem. But my question is about `(?i:if\s?\([\d\w]\s*?[=<>~])`.

**Comment by theseion** [comment]
Nice catch @NiceYouKnow. The exact query syntax isn't relevant. The issue is that by the rule could be circumvented by using a number > 9 in the comparison. 

I've moved the four statements into a new regex data file.

I also adjusted the fourth rule to fix the false negative (test included) and removed the unnecessary character class still left in the original suggestion (`[\w]`).

**Comment by dune73** [comment]
Closing this in favor of #2348. Thank you for reporting @NiceYouKnow and for providing a PR @theseion.
